compiler/rustc_errors/src/lib.rs

```diff
@@ -1518,14 +1513,24 @@ impl HandlerInner {
         if self.treat_err_as_bug() {
             match (
                 self.err_count() + self.lint_err_count,
+                self.delayed_bug_count(),
                 self.flags.treat_err_as_bug.map(|c| c.get()).unwrap_or(0),
             ) {
-                (1, 1) => panic!("aborting due to `-Z treat-err-as-bug=1`"),
-                (0 | 1, _) => {}
```

Comment:
This will literally never trigger -- because if the err count is 0 or 1, then for `self.treat_err_as_bug()` to have evaluated to true, then we must have `-Ztreat-err-as-bug=1`, which would have matched the previous arm.